oci/oci_transport.go

```diff
@@ -193,7 +193,11 @@ func (ref ociReference) ociLayoutPath() string {
 
 // blobPath returns a path for a blob within a directory using OCI image-layout conventions.
 func (ref ociReference) blobPath(digest string) string {
-	return filepath.Join(ref.dir, "blobs", strings.Replace(digest, ":", "-", -1))
+	pts := strings.SplitN(digest, ":", 2)
+	if len(pts) != 2 {
+		return ""
+	}
```

Comment:
… and a verbose error message (say "Internal error: unexpected digest reference”) would be nice.